File: java/templates/java-maven-spring-boot/src/main/java/com/example/restatestarter/HelloController.java
package com.example.restatestarter;

import com.example.restatestarter.Greeter.Greeting;
import dev.restate.client.Client;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

	private final Client restateClient;

    public HelloController(Client restateClient) {
        this.restateClient = restateClient;
    }

    @GetMapping("/")
	public String index() {
		return GreeterClient.fromClient(restateClient).greet(new Greeting("Alice")) + " from Spring Boot!";
	}

}

File: templates/kotlin-gradle-spring-boot/src/main/kotlin/com/example/restatestarter/HelloController.kt
package com.example.restatestarter

import dev.restate.sdk.client.Client
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

// Example how ot use the restate client within your REST Controllers

@RestController
class HelloController(private val restateClient: Client) {

  @GetMapping("/")
  suspend fun index(): String {
    return GreeterClient.fromClient(restateClient).greet("Francesco") + " from Spring Boot!"
  }
}